common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/protocol/PushBlockStream.java

```diff
@@ -99,10 +110,11 @@ public void encode(ByteBuf buf) {
 
   public static PushBlockStream decode(ByteBuf buf) {
     String appId = Encoders.Strings.decode(buf);
+    int attemptId = buf.readInt();
```

Comment:
`PushBlockStream` is a public class. I'm afraid this would have the compatibility issue. cc @mridulm

Reply:
PushBlockStream is introduced with push-based shuffle which is not complete yet. The users will not have been able to use it because a number of necessary pieces were not added.

Reply:
Technically, this is a breaking change, since it was introduced in 3.1
But given it cant be used until push based shuffle can be enabled, this should be fine.
Thoughts @Ngone51 ?

Reply:
sgtm